### PR TITLE
Add a lobby option in D2k to disable Concretes.

### PR DESCRIPTION
**Pull request by MustaphaTR**
PChote gave support on Discord when i said
> The issue is not using them doesn't give you much of a penalty, so you are better off not wasting time on them. Ideally you would only want concrete under the Wind Traps, since those give half power without concrete.
I would like to see concrete being a lobby option tbh. I think it was planned to be in the original, since there are unused UI icons for it. I think the button is in too in the UI code, but it is out of the screen and if you move it, the button doesn't do anything.

so i ended up implementing it. The name of the checkbox can be changed, i couldn't think of something better.

### CODE REVIEW

File: OpenRA.Mods.D2k/Traits/Buildings/D2kActorPreviewPlaceBuildingPreview.cs
Comment (pchote) [inline OpenRA.Mods.D2k/Traits/Buildings/D2kActorPreviewPlaceBuildingPreview.cs]: This is only used here in the constructor, so this can be `var techTree = ` and the readonly var above removed.

File: OpenRA.Mods.D2k/Traits/Buildings/D2kActorPreviewPlaceBuildingPreview.cs
Comment (pchote) [inline OpenRA.Mods.D2k/Traits/Buildings/D2kActorPreviewPlaceBuildingPreview.cs]: Minor naming nit: IMO the check below would be clearer if we instead called this `checkUnsafeTiles`.

**Review by pchote — DISMISSED** [review]
LGTM :+1:

Just a couple of minor code style nits that we should fix before merging:

**Comment by MustaphaTR** [comment]
I forgot to say that i updated this yesterday.

File: OpenRA.Mods.Common/Traits/DamagedByTerrain.cs
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/DamagedByTerrain.cs:60]: Shouldn't this be changed to `TraitEnabled` to avoid issues with not initialising `damageThreshold`?

Comment (MustaphaTR) [reply OpenRA.Mods.Common/Traits/DamagedByTerrain.cs:60]: Not sure why it is in AddedToWorld at the first place. I would expect it to be on Created before and if we want it to reset to Threshold every time trait is enabled, makes sense for it to be on TraitEnabled. Could also change the trait to PausibleConditional so pausing don't recheck the StartOnThreshold, but disabling does.

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/DamagedByTerrain.cs:60]: I don't know what the usecase for making it pausable would be, so am indifferent on that. Thinking about it, `Created` might not work for the same reason as just `TraitEnabled`: What if the actor is created, but not added to the world directly? We probably need a combination of `Created`/`AddedToWorld` and `TraitEnabled` to cover the cases of a) actor created but only later added to the world, b) actor created and added to the world, but trait enabled later. (?)

I suppose we could get away with keeping the calculations in `AddedToWorld` but only deal the initial damage when the trait is not disabled.

Comment (MustaphaTR) [reply OpenRA.Mods.Common/Traits/DamagedByTerrain.cs:60]: Can an actor that isn't in world not get damaged? Why would you want to delay this until it is added to world? Currently with AddedToWorld it would deal the damage every time a unit is removed and added back to world, i don't think this would be intended.

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/DamagedByTerrain.cs:60]: It doesn't really occupy space when not in the world, so damaging on foundation doesn't make sense.

Comment (MustaphaTR) [reply OpenRA.Mods.Common/Traits/DamagedByTerrain.cs:60]: Yeah, you are right. I forgot which trait we were dealing with. Still no idea what exactly to do tho.

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/DamagedByTerrain.cs:60]: I'll file a follow-up PR, as I noticed it's a pre-existing bug and it's kinda unfair to block this PR (and the milestone) over it.

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e8d9a636a02bfe6db659f0ee25280eba9e41d563...841e349fe0551f54cf552528d714ffe45fac3564)

**Pull request merged**
